common/src/Common/Wallet.hs

```diff
@@ -274,54 +302,239 @@ instance FromJSON UnfinishedCrossChainTransfer where
-    [] -> Nothing
-  ) ''Account
-
-instance ToJSON key => ToJSON (Account key) where
-  toJSON a = object $ catMaybes
-    [ Just $ "name" .= _account_name a
-    , Just $ "key" .= _account_key a
-    , Just $ "chain" .= _account_chainId a
-    , Just $ "network" .= _account_network a
-    , Just $ "notes" .= _account_notes a
-    , ("balance" .=) <$> _account_balance a
-    , ("unfinishedCrossChainTransfer" .=) <$> _account_unfinishedCrossChainTransfer a
+instance ToJSON AccountInfo where
+  toJSON a = object
+    [ "balance" .= toJSON (_accountInfo_balance a)
+    , "unfinishedCrossChainTransfer" .= toJSON (_accountInfo_unfinishedCrossChainTransfer a)
+    , "hidden" .= toJSON (_accountInfo_hidden a)
     ]
 
-instance FromJSON key => FromJSON (Account key) where
-  parseJSON = withObject "Account" $ \o -> do
-    name <- o .: "name"
+instance FromJSON AccountInfo where
+  parseJSON = withObject "AccountInfo" $ \o -> do
+    balance <- o .: "balance"
+    unfinishedCrossChainTransfer <- lenientLookup o "unfinishedCrossChainTransfer"
+    hidden <- o .: "hidden"
+    pure $ AccountInfo
+      { _accountInfo_balance = balance
+      , _accountInfo_unfinishedCrossChainTransfer = unfinishedCrossChainTransfer
+      , _accountInfo_hidden = hidden
+      }
+
+data VanityAccount = VanityAccount
+  { _vanityAccount_key :: PublicKey -- TODO should be KeySet, but that's a huge change.
+  , _vanityAccount_notes :: AccountNotes
+  , _vanityAccount_info :: AccountInfo
+  }
+
+instance ToJSON VanityAccount where
+  toJSON as = object
+    [ "key" .= toJSON (_vanityAccount_key as)
+    , "notes" .= toJSON (_vanityAccount_notes as)
+    , "info" .= toJSON (_vanityAccount_info as)
+    ]
+
+instance FromJSON VanityAccount where
+  parseJSON = withObject "VanityAccount" $ \o -> do
     key <- o .: "key"
-    chain <- o .: "chain"
-    network <- o .: "network"
     notes <- o .: "notes"
-    balance <- o .:? "balance"
-    unfinishedCrossChainTransfer <- lenientLookup o "unfinishedCrossChainTransfer"
-    pure $ Account
-      { _account_name = name
-      , _account_key = key
-      , _account_chainId = chain
-      , _account_network = network
-      , _account_notes = notes
-      , _account_balance = balance
-      , _account_unfinishedCrossChainTransfer = unfinishedCrossChainTransfer
+    info <- o .: "info"
+    pure $ VanityAccount
+      { _vanityAccount_key = key
+      , _vanityAccount_notes = notes
+      , _vanityAccount_info = info
       }
 
+data NonVanityAccount = NonVanityAccount
+  { _nonVanityAccount_info :: AccountInfo
+  }
+
+instance ToJSON NonVanityAccount where
+  toJSON as = object
+    [ "info" .= toJSON (_nonVanityAccount_info as)
+    ]
+
+instance FromJSON NonVanityAccount where
+  parseJSON = withObject "NonVanityAccount" $ \o -> do
+    info <- o .: "info"
+    pure $ NonVanityAccount
+      { _nonVanityAccount_info = info
+      }
+
+newtype AccountStorage = AccountStorage { unAccountStorage :: Map NetworkName Accounts }
```

Comment:
Is there a reason (besides the code currently being extremely biased towards it, I mean) for `NetworkName` to be outside `Accounts`, but `ChainId` be inside the maps?

Reply:
Only the structure of the JSON and to reflect the split a little more